controllers/dopplersecret_controller.go

```diff
@@ -83,6 +76,29 @@ func (r *DopplerSecretReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}, nil
 	}
 
+	// If omitted, the default namespace for references is the DopplerSecret's namespace
+	tokenSecretRefNamespace := dopplerSecret.Spec.TokenSecretRef.Namespace
+	if tokenSecretRefNamespace == "" {
+		tokenSecretRefNamespace = dopplerSecret.Namespace
+	}
+	managedSecretRefNamespace := dopplerSecret.Spec.ManagedSecretRef.Namespace
+	if managedSecretRefNamespace == "" {
+		managedSecretRefNamespace = dopplerSecret.Namespace
+	}
+
+	if ownNamespace == dopplerSecret.Namespace {
+		log.Info("Reconciling dopplersecret in operator namespace, references can be in any namespace.")
+	} else if dopplerSecret.Namespace == tokenSecretRefNamespace && dopplerSecret.Namespace == managedSecretRefNamespace {
+		log.Info("Reconciling dopplersecret in non-operator namespace, all references are in the same namespace as the dopplersecret.")
+	} else {
+		p1 := fmt.Sprintf("cannot reconcile dopplersecret (%v/%v) in a namespace different from the operator (%v)", dopplerSecret.Namespace, dopplerSecret.Name, ownNamespace)
+		p2 := fmt.Sprintf("unless all secret references [(%v/%v), (%v/%v)] are also in the dopplersecret's namespace", tokenSecretRefNamespace, dopplerSecret.Spec.TokenSecretRef.Name, managedSecretRefNamespace, dopplerSecret.Spec.ManagedSecretRef.Name)
+		log.Error(fmt.Errorf("%v %v", p1, p2), "")
```

Comment:
Nit: Why use the temporary variables p1 and p2 at all?

Reply:
Good callout -- I just broke these into two parts to break the line and to make the parameter formatting easier to read.